src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.PublishVerticalManifest/PublishVerticalManifest.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.SourceBuild.Tasks.PublishVerticalManifest
+{
+    public class PublishVerticalManifest : Task
+    {
+        [Required]
+        public ITaskItem[] AssetManifestPath { get; set; }
+        [Required]
+        public string VerticalAssetManifestOutputPath { get; set; }
+
+        public override bool Execute()
+        {
+            var assetManifestXmls = AssetManifestPath.Select(xmlPath => XDocument.Load(xmlPath.ItemSpec)).ToList();
+
+            VerifyAssetManifests(assetManifestXmls);
+
+            var rootName = assetManifestXmls.First().Root.Name;
+            var rootAttributes = assetManifestXmls.First().Root.Attributes();
+            var packageElements = new List<XElement>();
+            var blobElements = new List<XElement>();
+
+            foreach (var assetManifestXml in assetManifestXmls)
+            {
+                packageElements.AddRange(assetManifestXml.Descendants("Package"));
+                blobElements.AddRange(assetManifestXml.Descendants("Blob"));
+            }
+            
+            packageElements = packageElements.OrderBy(packageElement => packageElement.Attribute("Id").Value).ToList();
+            blobElements = blobElements.OrderBy(blobElement => blobElement.Attribute("Id").Value).ToList();
+
+            var verticalManifest = new XDocument(new XElement(rootName, rootAttributes, packageElements, blobElements));
+
+            File.WriteAllText(VerticalAssetManifestOutputPath, verticalManifest.ToString());
+
+            return true;
+        }
+
+        private static void VerifyAssetManifests(List<XDocument> assetManifestXmls)
+        {
+            if (assetManifestXmls.Count == 0)
+            {
+                throw new ArgumentException("No asset manifests were provided.");
+            }
+
+            var rootAttributes = assetManifestXmls
+                .First()
+                .Root
+                .Attributes()
+                .Select(attribute => attribute.ToString())
+                .ToHashSet();
+
+            if (assetManifestXmls.Skip(1).Any(assetManifestXml => 
+                    !assetManifestXml.Root.Attributes().All(attribute => 
+                        rootAttributes.Contains(attribute.ToString()))))
```

Comment:
You may want to add one more check here, which is that the attr count is the same between assetManifestXml and rootAttributes. Otherwise the first asset manifest xml could contain extra attributes and the check would still pass.

Reply:
that's a good point, I'll add it. Think we'll also have to ignore the `AzureDevOpsBuildNumber` and `BuildId` attributes. For some reason, those are not the same in all repos in a vertical, which is weird, we're using the `$(BUILD_BUILDNUMBER)` env variable as the value. What's even weirder is that, for example, in all builds, and all legs, emsdk `AssetManifests` will have the same `BuildId`.

Reply:
Added the check